packages/client/lib/CredentialOffer/IssuanceInitiationClient.ts

```diff
@@ -0,0 +1,92 @@
+import { EndpointMetadata, IssuanceInitiationRequestPayload, IssuanceInitiationWithBaseUrl, OpenId4VCIVersion } from '@sphereon/openid4vci-common';
+import Debug from 'debug';
+
+import { MetadataClient } from '../MetadataClient';
+import { convertJsonToURI, convertURIToJsonObject } from '../functions';
+
+import { CredentialIssuanceClient } from './index';
+
+const debug = Debug('sphereon:openid4vci:initiation');
+
+export class IssuanceInitiationClient implements CredentialIssuanceClient {
+  public readonly _version: OpenId4VCIVersion;
+  private readonly _issuanceInitiationWithBaseUrl: IssuanceInitiationWithBaseUrl;
+
+  public constructor(issuanceInitiationWithBaseUrl: IssuanceInitiationWithBaseUrl) {
+    this._version = OpenId4VCIVersion.VER_9;
+    this._issuanceInitiationWithBaseUrl = issuanceInitiationWithBaseUrl;
+  }
+
+  public static fromURI(issuanceInitiationURI: string): IssuanceInitiationClient {
+    debug(`issuance initiation URI: ${issuanceInitiationURI}`);
+    if (!issuanceInitiationURI.includes('?')) {
+      debug(`Invalid issuance initiation URI: ${issuanceInitiationURI}`);
+      throw new Error('Invalid Issuance Initiation Request Payload');
+    }
+    const baseUrl = issuanceInitiationURI.split('?')[0];
+    const issuanceInitiationRequest = convertURIToJsonObject(issuanceInitiationURI, {
+      arrayTypeProperties: ['credential_type'],
+      requiredProperties: ['issuer', 'credential_type'],
+    }) as IssuanceInitiationRequestPayload;
+
+    return new IssuanceInitiationClient({
+      baseUrl,
+      issuanceInitiationRequest,
+    });
+  }
+
+  public static toURI(issuanceInitiationWithBaseUrl: IssuanceInitiationWithBaseUrl): string {
```

Comment:
Why would this be a static on a class with state that is not responsible for it to begin with?